pcsx2/x86/iR3000A.cpp

```diff
@@ -1021,11 +1029,167 @@ void rpsxBREAK()
 	//if (!psxbranch) psxbranch = 2;
 }
 
+void psxDynarecCheckBreakpoint()
+{
+	u32 pc = psxRegs.pc;
+	if (CBreakPoints::CheckSkipFirst(BREAKPOINT_IOP, pc) == pc)
+		return;
+
+	int bpFlags = psxIsBreakpointNeeded(pc);
+	bool hit = false;
+	//check breakpoint at current pc
+	if (bpFlags & 1) {
+		auto cond = CBreakPoints::GetBreakPointCondition(BREAKPOINT_IOP, pc);
+		if (cond == NULL || cond->Evaluate()) {
+			hit = true;
+		}
+	}
+	//check breakpoint in delay slot
+	if (bpFlags & 2) {
+		auto cond = CBreakPoints::GetBreakPointCondition(BREAKPOINT_IOP, pc + 4);
+		if (cond == NULL || cond->Evaluate())
+			hit = true;
+	}
+
+	if (!hit)
+		return;
+
+	CBreakPoints::SetBreakpointTriggered(true);
+	GetCoreThread().PauseSelf();
+	iopBreakpoint = true;
+}
+
+void psxDynarecMemcheck()
+{
+	u32 pc = psxRegs.pc;
+	if (CBreakPoints::CheckSkipFirst(BREAKPOINT_IOP, pc) == pc)
+		return;
+
+	CBreakPoints::SetBreakpointTriggered(true);
+	GetCoreThread().PauseSelf();
```

Comment:
After hours of debugging, I found that you needed to set this is `GetCoreThread().PauseSelfDebug();` 
This fixes the  `Step over/into/out` buttons and matches the way the R5900 dynarec functions
Currently, when you use the stepping buttons on the R3000 the `break/run` button isn't properly updated and the temporary breakpoints aren't removed.